test/integration/sellers-api-client.test.ts

```diff
@@ -0,0 +1,74 @@
+import { AssumeRoleCommand, STSClient } from '@aws-sdk/client-sts'
+import axios from 'axios'
+
+import { SellersApiClient } from '../../src/api-clients/sellers-api-client'
+import * as environment from '../environment'
+
+interface TokenResponse {
+  /* eslint-disable camelcase */
+  access_token: string
+  expires_in: number
+  refresh_token: string
+  /* eslint-enable camelcase */
+}
+
+describe(`${SellersApiClient.name}`, () => {
+  it(
+    'should return the list of marketplaces',
+    async () => {
+      expect.assertions(1)
+
+      const { data: tokens } = await axios.post<TokenResponse>(
+        'https://api.amazon.com/auth/o2/token',
+        {
+          grant_type: 'refresh_token',
+          client_id: environment.CLIENT_ID,
+          client_secret: environment.CLIENT_SECRET,
+          refresh_token: environment.REFRESH_TOKEN,
+        },
+      )
+
+      const sts = new STSClient({
+        region: environment.API_REGION,
+        credentials: {
+          accessKeyId: environment.AWS_ACCESS_KEY_ID,
+          secretAccessKey: environment.AWS_SECRET_ACCESS_KEY,
+        },
+      })
+
+      const { Credentials } = await sts.send(
+        new AssumeRoleCommand({
+          RoleArn: environment.ROLE_ARN,
+          RoleSessionName: 'selling-partner-api-axios',
+        }),
+      )
+
+      const client = new SellersApiClient({
+        basePath: environment.API_URL,
+        accessToken: tokens.access_token,
+        apiModelProperties: {
+          region: environment.API_REGION,
+          credentials: {
+            // TODO: fix these warnings
+            accessKeyId: Credentials!.AccessKeyId || '',
+            secretAccessKey: Credentials!.SecretAccessKey || '',
+            sessionToken: Credentials!.SessionToken || '',
+          },
+        },
+      })
+
+      /**
+       * TODO: response does not match types.
+       */
+
+      const response = await client.getMarketplaceParticipations()
```

Comment:
Here, `response` seems to be an `AxiosResponse`, but the API types are returning `response.data`. There is a type mismatch somewhere.

Reply:
@moltar : 
It seem to be not able to override response type when we unwrap `AxiosResponse` in axios interceptors.

In perhaps, we need to override `AxiosResponse`:
```
import axios from 'axios'

declare module 'axios' {
  export interface AxiosResponse<T = any> extends Promise<T> {}
}
```

But I don't think that is good idea. Therefore, i removed unwrapped `AxiosResponse` in axios interceptors.

Ref: https://github.com/axios/axios/issues/1510#issuecomment-448201698